### PR TITLE
CLOUD-14: Add documentation

### PR DESCRIPTION
**Pull request by pdudits**
Added javadoc to Configuration and Deployment process.

Made micro configuration for tests platform independent.

### CODE REVIEW

File: deployer/deployment-controller/src/main/java/fish/payara/cloud/deployer/process/Configuration.java [hunk 1]
```diff
@@ -188,23 +291,47 @@ public boolean isUpdated(String key) {
...
+         * @param converter converter to invoke
+         * @param <T> target type of t
+         * @return Non-empty optional if value is defined and conversion succeeds. Empty optional otherwise
+         */
         public <T> Optional<T> convert(Function<? super String,T> converter) {
             try {
                 return getValue(validationContext).map(converter);
-            } catch (RuntimeException re) {
-                addValidationError(re.getMessage());
+            } catch (Throwable t) {
```
Comment (Cousjava) [inline deployer/deployment-controller/src/main/java/fish/payara/cloud/deployer/process/Configuration.java]: Why is this now a throwable? What non-exception can be thrown here that needs to be caught?

File: deployer/deployment-controller/src/test/java/fish/payara/cloud/deployer/process/ConfigurationTest.java [hunk 2]
```diff
@@ -266,7 +265,7 @@ protected void checkUpdate(UpdateContext context) {
             var optional = context.key("optional").check(this::startsWithO);
 
             if (optional.isPresent() && optional.get().equals("odd") && context.updatedKeys().size() % 2 == 1) {
-                context.setTopLevelContext();
+                context.topLevel();
```
Comment (Cousjava) [inline deployer/deployment-controller/src/test/java/fish/payara/cloud/deployer/process/ConfigurationTest.java]: What is the reason for this change?

File: deployer/deployment-controller/src/main/java/fish/payara/cloud/deployer/process/Configuration.java (hunk 1 above)
Comment (pdudits) [reply deployer/deployment-controller/src/main/java/fish/payara/cloud/deployer/process/Configuration.java]: I considered `AssertionError`

File: deployer/deployment-controller/src/test/java/fish/payara/cloud/deployer/process/ConfigurationTest.java (hunk 2 above)
Comment (pdudits) [reply deployer/deployment-controller/src/test/java/fish/payara/cloud/deployer/process/ConfigurationTest.java]: Because it's a sibling to method `key(String)`, and it's fluent one, therefore calling it `set` feels unnatural. (The API changed as well).
